### ISSUE TITLE
HTML formating in table cells <td>

### ISSUE THREAD

**Issue by tplevrak** [lazymofo/datagrid#71]
Dear Ian,

I was trying to put a couple of <br> tags in a table cell so that the text will change line but it didn't work.

Then i tried to have a nested table in a table cell but again it does not work.

It seems as it escapes the tag opening and closing characters, (< and >), so the browser does not regard them like html code, rather than a simple string.

Is there a way that i can have some html formatting in a table cell ?

Best Regards

**Comment by lazymofo** [comment]
Everything from the user/database needs to be escaped to prevent XSS
attacks.
If you want to add html in the output you need to use a custom output
control.


$lm->grid_output_control['weird_data'] = array("type" =>
"my_user_function");

function my_user_function($column_name, $value, $command, $called_from){

    // $column_name: field name
    // $value: field value
    // $command: array defined in the control, or string with legacy
syntax; rarely needed
    // $called_from: orgin of call; 'form', or 'grid'; rarely needed

    global $lm;
    $val = $lm->clean_out($value);
    return "<b>$val</b>";

}

**Comment by tplevrak** [comment]
Dear Ian,

Thank you again for your time and your solution.

I did not know about XSS attacks but i thought something like that was the reason.

Anyway, your solution is perfect, i have already implemented it in my application.

I did not want to do anything fancy, i just wanted to show 3 fields with codes of a product in the same table cell so as not to waste horizontal space for three different columns. No input or alterations of the values needed.

Your solution is perfect.

Thank you once again.

Best Regards
